control-plane-operator/controllers/hostedcontrolplane/v2/kas/kubeconfig.go

```diff
@@ -116,7 +117,31 @@ func adapExternalAdminKubeconfigSecret(cpContext component.WorkloadContext, secr
 	return nil
 }
 
+func adaptCustomAdminKubeconfigSecret(cpContext component.WorkloadContext, secret *corev1.Secret) error {
+	hcp := cpContext.HCP
+
+	if len(hcp.Spec.KubeAPIServerDNSName) > 0 {
+		secret.Name = hcp.Spec.KubeAPIServerDNSName
```

Comment:
why are we changing the secret name here?

Reply:
nice catch 🙏